File: c#/gcd-lcm.cs
using System.IO;
using System;

class Program
{
    static int gcd(int a,int b){
    	int m;
    	while(b!=0){
    		m=a%b;
    		a=b;
    		b=m;
    	}
    	return a;
    }
    
    static int lcm(int a,int b){
    	int m = (a*b)/gcd(a,b);
    	return m;
    }
    static void Main()
    {
        Console.WriteLine(gcd(664,414));
        Console.WriteLine(lcm(664,414));
    }
}

File: typescript/gcd-lcm.ts
function gcd(a:number, b:number){
	let m:number;
	while(b!=0){
		m=a%b;
		a=b;
		b=m;
	}
	return a;
}

function lcm(a:number, b:number){
	let m:number = (a*b)/gcd(a,b);
	return m;
}

console.log(gcd(664,414));
console.log(lcm(664,414));